src/core/thread/mesh_forwarder.cpp

```diff
@@ -220,7 +220,7 @@ void MeshForwarder::UpdateIndirectMessages(void)
     {
         Child *child = &children[i];
 
-        if (child->mState == Child::kStateValid || child->mQueuedIndirectMessageCnt == 0)
+        if (child->IsInValidOrRestoringState() || child->mQueuedIndirectMessageCnt == 0)
```

Comment:
nit: `IsInValid` reads like `IsInvalid`.  Maybe it would read better as `IsStateValidOrRestoring()`.